publish_courses/api.py

```diff
@@ -152,178 +130,119 @@ def create(self, request, *args, **kwargs) -> Response:
             raise DRFValidationError('Both account and term are required')
 
         # For the moment, only the current school account can be operated on.
-        if not account_sis_id[len('school:'):] == account:
+        if not account_sis_id.removeprefix('school:') == account:
             raise PermissionDenied
 
         selected_courses = self.request.data.get('course_list')
         account = f'sis_account_id:school:{account}'
         term = f'sis_term_id:{term}'
 
-        canvas_courses = self.get_canvas_courses(account, term)
         if not selected_courses:
-            selected_courses = canvas_courses
+            courses = _get_courses(account, term)
+            # Filter unpublished courses.
+            selected_courses = [cs_dict['id'] for cs_dict in courses if cs_dict['workflow_state'] == 'unpublished']
+
+        logger.info(f"{len(selected_courses)} selected courses for {account} and term {term}.",
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

You have a log statement that might use unsanitized input. Depending on the context, this could result in:

1. A log injection attack that breaks log integrity, forges log entries, or bypasses monitors that use the logs. To increase the security of your code, sanitize your inputs before logging them. [Learn more](https://cwe.mitre.org/data/definitions/117.html)

2. A sensitive information leak that exposes users&#39; credentials, private information, or identifying information to an attacker. To preserve privacy in your code, redact sensitive user information before logging it. [Learn more](https://cwe.mitre.org/data/definitions/532.html)

Similar issue at line numbers 230 and 247.